search/intermediate_result.cpp

```diff
@@ -277,9 +275,8 @@ bool PreResult2::EqualLinearTypesF::operator() (PreResult2 const & r1, PreResult
 {
   // Note! Do compare for distance when filtering linear objects.
   // Otherwise we will skip the results for different parts of the map.
-  return (r1.m_geomType == feature::GEOM_LINE &&
-          r1.IsEqualCommon(r2) &&
-          PointDistance(r1.GetCenter(), r2.GetCenter()) < DIST_SAME_STREET);
+  return (r1.m_geomType == feature::GEOM_LINE && r1.IsEqualCommon(r2) &&
```

Comment:
Please, take a look at the CL title - "Fixed search in viewport filtering", not "... filtering and whole legacy code refactoring".